routes/api.php

```diff
@@ -42,3 +59,29 @@
         Route::delete('/{user}', DeleteUserController::class)
             ->whereNumber('user');
     });
+
+Route::prefix('doctors')
+    ->group(static function(): void {
+        Route::get('{doctor}', GetDoctorController::class)
+            ->withTrashed()
+            ->whereNumber('doctor');
+        Route::get('/', ListDoctorController::class);
+        Route::post('/', StoreDoctorController::class);
+        Route::put('/{doctor}', UpdateDoctorController::class)
+            ->whereNumber('doctor');
+        Route::delete('/{doctor}', DeleteDoctorController::class)
+            ->whereNumber('doctor');
```

Comment:
```suggestion
        Route::get('/', ListDoctorController::class);
        Route::post('/', StoreDoctorController::class);

        Route::prefix('{doctor}')
            ->whereNumber('doctor')
            ->group(static function () {
                Route::get(', GetDoctorController::class)->withTrashed();
                Route::put('/', UpdateDoctorController::class);
                Route::delete('/', DeleteDoctorController::class);
            });
```
You could group the route model-bound routes so that you apply the `whereNumber` once.

Reply:
That's a great advice